src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -141,6 +163,46 @@ public int MaxDepth
             }
         }
 
+        /// <summary>
+        /// Specifies the policy used to convert a property's name on an object to another format, such as camel-casing.
+        /// The resulting property name is expected to match the JSON payload during deserialization, and
+        /// will be used when writing the property name during serialization.
+        /// </summary>
+        /// <remarks>
+        /// The policy is not used for properties that have a <see cref="JsonPropertyNameAttribute"/> applied.
+        /// This property can be set to <see cref="JsonNamingPolicy.CamelCase"/> to specify a camel-casing policy.
+        /// </remarks>
+        public JsonNamingPolicy PropertyNamingPolicy
+        {
+            get
+            {
+                return _jsonPropertyNamingPolicy;
+            }
+            set
+            {
+                VerifyMutable();
+                _jsonPropertyNamingPolicy = value;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether a property's name uses a case-insensitive comparison during deserialization.
+        /// The default value is false.
+        /// </summary>
+        /// <remarks>There is a performance cost associated when the value is true.</remarks>
```

Comment:
Do you have a sense of what the cost is? I am wondering if this is worth mentioning.

Reply:
Not yet. I'll measure once I have a build.